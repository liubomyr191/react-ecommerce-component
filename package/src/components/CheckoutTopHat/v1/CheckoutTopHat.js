import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { applyTheme } from "../../../utils";

const TopHatContainer = styled.div`
  font-family: ${applyTheme("font_family")};
  background-color: ${applyTheme("color_black05")};
  display: flex;
  height: 30px;
  justify-content: center;
  width: 100%;
`;

const TopHatMessage = styled.div`
  display: flex;
  align-items: center;
  color: ${applyTheme("color_coolGrey500")};
  font-size: ${applyTheme("font_size_small")};
  font-weight: ${applyTheme("font_weight_bold")};
`;

class CheckoutTopHat extends Component {
  static propTypes = {
    checkoutMessage: PropTypes.string
  };

  render() {
    const { checkoutMessage } = this.props;

    if (checkoutMessage) {
      return (
        <TopHatContainer>
          <TopHatMessage>{checkoutMessage}</TopHatMessage>
        </TopHatContainer>
      );
    }

    return null;
  }
}

export default CheckoutTopHat;
