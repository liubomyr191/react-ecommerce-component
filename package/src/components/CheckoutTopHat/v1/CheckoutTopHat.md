### CheckoutTopHat

### Overview
Displays a message in a "Top hat" at the top of the checkout page.

#### Basic usage
```jsx
<CheckoutTopHat
  checkoutMessage="Free Shipping + Free Returns"
/>
```
