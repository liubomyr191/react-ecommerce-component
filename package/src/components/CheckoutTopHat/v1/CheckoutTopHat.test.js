import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import CheckoutTopHat from "./CheckoutTopHat";

test("basic snapshot", () => {
  const component = renderer.create(<CheckoutTopHat />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
