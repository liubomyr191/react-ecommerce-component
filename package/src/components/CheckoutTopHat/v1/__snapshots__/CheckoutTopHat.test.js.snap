// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Top Hat with message 1`] = `
.c0 {
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  background-color: #f5f5f5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 35px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #3c3c3c;
  font-size: 14px;
  font-weight: 700;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Free Shipping + Free Returns
  </div>
</div>
`;

exports[`render nothing when no message is present 1`] = `null`;
